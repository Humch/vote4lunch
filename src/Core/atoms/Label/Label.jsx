/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';

import PropTypes from 'prop-types';

const Label = ({
  text,
  forProp,
  color,
}) => {
  const textColor = color ? ` text-${color}` : '';
  const classList = `${textColor}`;

  return (
    <label className={classList} htmlFor={forProp}>{text}</label>
  );
};

Label.propTypes = {
  text: PropTypes.string.isRequired,
  forProp: PropTypes.string.isRequired,
  color: PropTypes.string,
};

export default Label;
