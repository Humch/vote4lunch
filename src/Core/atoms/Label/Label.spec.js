/* eslint-disable no-undef */
import React from 'react';

import renderer from 'react-test-renderer';


import Label from './Label';


describe('Label Snapshot', () => {
  const text = 'Le superbe titre';
  const forProp = 'leLabel';
  const color = 'white';

  it('capturing Snapshot of Label with text and forProp only', () => {
    const renderedValue = renderer.create(
      <Label
        text={text}
        forProp={forProp}
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  it('capturing Snapshot of Label with text, forProp and color', () => {
    const renderedValue = renderer.create(
      <Label
        text={text}
        forProp={forProp}
        color={color}
      />
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
