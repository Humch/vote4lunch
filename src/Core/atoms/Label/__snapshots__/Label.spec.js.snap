// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Snapshot capturing Snapshot of Label with text and forProp only 1`] = `
<label
  className=""
  htmlFor="leLabel"
>
  Le superbe titre
</label>
`;

exports[`Label Snapshot capturing Snapshot of Label with text, forProp and color 1`] = `
<label
  className=" text-white"
  htmlFor="leLabel"
>
  Le superbe titre
</label>
`;
